pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -222,68 +221,66 @@ def set_config(self, cfg: SimConfig) -> None:
+                self.basis_name == "all" and self.dim == 3
+            ):  # three-level system
+                k = np.sqrt(prob * (1 - prob) ** (n - 1))
+                # m_0 = np.sqrt(1 - prob) * qutip.qeye(3)
+                for i in range(3):
+                    ket_ = qutip.basis(3, i)
+                    kraus_ops.append(k * ket_ * ket_.dag())
+            else:  # two-level system
+                prob = prob / 2
+                k = np.sqrt(prob * (1 - prob) ** (n - 1))
+                kraus_ops.append(k * qutip.sigmaz())
             self._collapse_ops += [
                 np.sqrt((1 - prob) ** n)
                 * qutip.tensor([self.op_matrix["I"] for _ in range(n)])
             ]
-            kraus_ops.append(k * qutip.sigmaz())
 
         if "depolarizing" in self.config.noise:
-            if self.basis_name == "digital" or self.basis_name == "all":
-                # Go back to previous config
-                self.set_config(prev_config)
-                raise NotImplementedError(
-                    "Cannot include depolarizing "
-                    + "noise in digital- or all-basis."
-                )
-            # Probability of error occurrence
-
-            prob = self.config.depolarizing_prob / 4
+            prob = self.config.depolarizing_prob
             n = self._size
             if prob > 0.1 and n > 1:
                 warnings.warn(
                     "The depolarizing model is a first-order approximation"
-                    f" in the depolarizing probability. p = {4*prob}"
+                    f" in the depolarizing probability. p = {prob}"
                     " is too large for realistic results.",
                     stacklevel=2,
                 )
-
-            k = np.sqrt((prob) * (1 - 3 * prob) ** (n - 1))
+            if (
+                self.basis_name == "all" and self.dim == 3
+            ):  # three-level system
+                prob = prob / 3
+
+                k = np.sqrt((prob) * (1 - 3 * prob) ** (n - 1))
+                for i in range(3):
+                    for j in range(3):
+                        ket_ = qutip.basis(3, i)
+                        bra_ = qutip.basis(3, j).dag()
+                        kraus_ops.append(k * ket_ * bra_)
+
+            else:  # two-level system
+                prob = prob / 4
+                k = np.sqrt((prob) * (1 - 3 * prob) ** (n - 1))
+                kraus_ops.append(k * qutip.sigmax())
+                kraus_ops.append(k * qutip.sigmay())
+                kraus_ops.append(k * qutip.sigmaz())
             self._collapse_ops += [
                 np.sqrt((1 - 3 * prob) ** n)
                 * qutip.tensor([self.op_matrix["I"] for _ in range(n)])
             ]
-            kraus_ops.append(k * qutip.sigmax())
-            kraus_ops.append(k * qutip.sigmay())
-            kraus_ops.append(k * qutip.sigmaz())
 
         if "eff_noise" in self.config.noise:
-            if self.basis_name == "digital" or self.basis_name == "all":
```

Comment:
In this case you should now check if the dimension of the Kraus operators is compatible with `self.dim`

Reply:
Here did you mean the `eff_noise_opers`? Something like:
```
if not np.all(op.shape == (self.dim, self.dim) for op in self.config.eff_noise_opers):
    # Go back to previous config
    self.set_config(prev_config)
    raise NotImplementedError(
        "Cannot mix noise operator dimensions."
    )
```

Reply:
Not sure it's necessary to revert the config. You are already checking that all the operators have the same dimension inside `NoiseModel`, here you just have to check that it is compatible with `self.dim` (eg the first operator has dimension `(self.dim, self.dim)`). 

I would change the error message to inform that the `eff_noise_opers` dimensions are incompatible with the state's dimensions. Including what those dimensions are in the error message is even better.